docs/general/developer-resources/emerald-paratime/writing-dapps-on-emerald.md

```diff
@@ -614,6 +614,434 @@ share them with us at [#dev-support on our Community Slack channel][slack-devsup
 [metamask]: ../../manage-tokens/how-to-transfer-rose-into-emerald-paratime.mdx#verifying-rose-balance-on-emerald-paratime
 [slack-devsupport]: https://oasiscommunity.slack.com/messages/dev-support
 
+## Create dApp on Emerald with Rust
+
+This tutorial assumes you have a working Rust development environment.
+
+### Setup for Rust
+
+In order to be able to compile smart contracts, first you must
+[install solc](https://docs.soliditylang.org/en/latest/installing-solidity.html#linux-packages).
+
+For the purpose of this tutorial, create a file called `Greeter.sol` with the following code:
+
+```js
+pragma solidity ^0.8.0;
+
+contract Greeter {
+    string private greeting;
+
+    constructor(string memory _greeting) {
+        greeting = _greeting;
+    }
+
+    function greet() public view returns (string memory) {
+        return greeting;
+    }
+
+    function setGreeting(string memory _greeting) public {
+        greeting = _greeting;
+    }
+}
+```
+
+The smart contract above can also be used in [the Go tutorial](#create-dapp-on-emerald-with-go).
+
+### Compile and deploy via Rust
+
+You will need to add the following dependencies to `Cargo.toml` in order to be able to compile
+and to deploy smart contracts:
+
+```
+[package]
+name = "sol-deploy"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+ethers = { git = "https://github.com/gakonst/ethers-rs/" }
+eyre = "0.6.8"
+serde_json = "1.0.81"
+hex = "0.4.3"
+tokio = { version = "1.17", features = [ "rt", "macros" ] }
```

Comment:
Let's unify `Cargo.toml` for compilation, deployment and testing smart contracts and implement the following:
- calling `cargo build` should compile the rust code and *also* call `solc` to compile the smart contract(s).
- implement proper Rust tests (deployment+calling) for the smart contract to support `cargo test`.
- for deploying only, a custom `cargo deploy` rule should be added.

The flow above is typical for hardhat, truffle and similar frameworks. Maybe something like this exists for Rust too?